lib/compilers/python.js

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2019, Sebastian Rath
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+const BaseCompiler = require('../base-compiler'),
+    argumentParsers = require("./argument-parsers");
+
+class PythonCompiler extends BaseCompiler {
+
+    constructor(compilerInfo, env) {
+        super(compilerInfo, env);
+        this.compiler.demangler = null;
+        this.demanglerClass = null;
+    }
+
+    // eslint-disable-next-line no-unused-vars
+    processAsm(result, filters) {
+        const lineRe = /^\s{0,4}([0-9]+)(.*)/;
+
+        const bytecodeLines = result.asm.split("\n");
+
+        var bytecodeResult = [];
+        var lastLineNo = null;
+        var sourceLoc = null;
+
+        bytecodeLines.forEach(line => {
+            var match = line.match(lineRe);
+
+            if (match) {
+                const lineno = parseInt(match[1]);
+                sourceLoc = {line: lineno, file: null};
+                lastLineNo = lineno;
+            } else if (!line) {
+                sourceLoc = {line: null, file: null};
+                lastLineNo = null;
+            } else {
+                sourceLoc = {line: lastLineNo, file: null};
+            }
+
+            bytecodeResult.push({text: line, source: sourceLoc});
+        });
+
+        return bytecodeResult;
+    }
+
+    optionsForFilter(filters, outputFilename) {
+        return ['-I',
+            '-c',
+            "import os, sys, dis;\
```

Comment:
I would rather have this as a wrapper script in `etc/scripts` since it may change depending on the python version